### PR TITLE
Remove Python2 from CI and code that branched on it

### PR DESCRIPTION
**Pull request by alex**
Refs #5533 (part 1)

### CODE REVIEW

**Comment by reaperhulk** [comment]
:its-happening:

File: setup.py [hunk 1]
```diff
@@ -63,8 +63,6 @@
             "Operating System :: POSIX :: Linux",
             "Operating System :: Microsoft :: Windows",
             "Programming Language :: Python",
-            "Programming Language :: Python :: 2",
-            "Programming Language :: Python :: 2.7",
             "Programming Language :: Python :: 3",
```
Comment (hugovk) [inline setup.py:66]: Could add this one:
```suggestion
            "Programming Language :: Python :: 3",
            "Programming Language :: Python :: 3 :: Only",
```

File: README.rst [hunk 2]
```diff
@@ -18,7 +18,7 @@ pyca/cryptography
 
 ``cryptography`` is a package which provides cryptographic recipes and
 primitives to Python developers.  Our goal is for it to be your "cryptographic
-standard library". It supports Python 2.7, Python 3.6+, and PyPy 5.4+.
+standard library". It supports Python 3.6+ and PyPy3 5.4+.
```
Comment (reaperhulk) [inline README.rst:21]: This is untrue. We only support PyPy3 with 3.6+ compat at this point (or at least we'll break PyPy3 <= that version when we add type annotations). This is probably 7.x?

File: .github/workflows/wheel-builder.yml [hunk 3]
```diff
@@ -57,10 +57,6 @@ jobs:
     strategy:
       matrix:
         PYTHON:
-          - VERSION: '2.7'
-            ABI_VERSION: '2.7'
-            DOWNLOAD_URL: 'https://www.python.org/ftp/python/2.7.17/python-2.7.17-macosx10.9.pkg'
-            BIN_PATH: '/Library/Frameworks/Python.framework/Versions/2.7/bin/python'
```
Comment (reaperhulk) [inline .github/workflows/wheel-builder.yml:63]: The regex match in the build the wheel bash for macOS can be removed now and we can define `ABI_VERSION` as just `cp36`.

File: setup.py [hunk 4]
```diff
@@ -79,13 +77,10 @@
             where="src", exclude=["_cffi_src", "_cffi_src.*"]
         ),
         include_package_data=True,
-        python_requires=(
-            ">=2.7,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,!=3.4.*,!=3.5.*"
-        ),
+        python_requires=">=3.6",
```
Comment (reaperhulk) [inline setup.py:81]: 🥳

File: README.rst (hunk 2 above)
Comment (reaperhulk) [reply README.rst]: Looks like 7.2

**Review by reaperhulk — APPROVED** [review]

File: README.rst (hunk 2 above)
Comment (hugovk) [reply README.rst]: * PyPy v7.0.0 was the first to support 3.6 as an alpha
https://morepypy.blogspot.com/2019/02/pypy-v700-triple-release-of-27-35-and.html

* PyPy v7.1 was the first to support 3.6 as a beta
https://morepypy.blogspot.com/2019/03/pypy-v71-released-now-uses-utf-8.html

* And PyPy v7.2 was the first with full 3.6 support
https://morepypy.blogspot.com/2019/10/pypy-v72-released.html

**Pull request merged**